src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -44,6 +59,36 @@ const AppFrame = ({ appUrl }: AppFrameProps): ReactElement => {
     setAppIsLoading(false)
   }, [appUrl, iframeRef, setAppIsLoading])
 
+  useEffect(() => {
+    const unsubscribe = txSubscribe(TxEvent.SAFE_APPS_REQUEST, async ({ txId, requestId }) => {
+      const currentRequestId = signMessageModalState.requestId || txModalState.requestId
+
+      if (txId && currentRequestId === requestId) {
+        const { detailedExecutionInfo } = await getTransactionDetails(chainId, txId)
+
+        if (isMultisigExecutionDetails(detailedExecutionInfo)) {
+          communicator?.send({ safeTxHash: detailedExecutionInfo.safeTxHash }, requestId)
+        }
+
+        txModalState.isOpen ? closeTxModal() : closeSignMessageModal()
+      }
+    })
+
+    return () => {
+      unsubscribe()
+    }
+  }, [chainId, closeTxModal, closeSignMessageModal, communicator, txModalState, signMessageModalState])
+
+  const onSafeAppsModalClose = () => {
+    if (txModalState.isOpen) {
+      communicator?.send(REJECT_TRANSACTION_MESSAGE, txModalState.requestId, true)
```

Comment:
But even creating another method, i would maintain the message as a parameter because perhaps i want to change it if the rejection is for another cause

Reply:
I see. Let's at least move the message constant itself to the communicator module then.

Reply:
make sense, done!